src/store/middleware/model-poller.ts

```diff
@@ -0,0 +1,138 @@
+import { Middleware } from "redux";
+import * as Sentry from "@sentry/browser";
+
+import {
+  actionsList,
+  storeLoginError,
+  updateControllerConnection,
+  updateJujuAPIInstance,
+  updatePingerIntervalId,
+} from "app/actions";
+import { isLoggedIn } from "app/selectors";
+import {
+  disableControllerUUIDMasking,
+  fetchAllModelStatuses,
+  fetchControllerList,
+  loginWithBakery,
+} from "juju";
+import { updateModelList } from "juju/actions";
+import { TSFixMe } from "@canonical/react-components";
+import { PayloadAction } from "@reduxjs/toolkit";
+
+// TODO: provide these types when the types are available from bakeryjs.
+// TODO: Import bakery instead of passing it as a param.
+type ControllerOptions = [
+  string,
+  TSFixMe,
+  TSFixMe,
+  boolean,
+  boolean | undefined
+];
+
+export const modelPollerMiddleware: Middleware = (reduxStore) => {
+  return (next) =>
+    async (
+      action: PayloadAction<{
+        controllers: ControllerOptions[];
+        isJuju: boolean;
+      }>
+    ) => {
+      if (action.type === actionsList.connectAndPollControllers) {
+        action.payload.controllers.forEach(async (controllerData) => {
```

Comment:
The contents of this loop is pretty much copied from `connectAndPollController` from [src/app/model-poller.js](https://github.com/canonical/jaas-dashboard/pull/1343/files#diff-70c5a8c9509f3d19534d6056094d41f47d6ae1f9a3b565679551acd7b42d0dd1L77).